tests/test_cli.py

```diff
@@ -235,3 +249,46 @@ def predict(self, context, model_input):  # pylint: disable=unused-variable
     assert scoring_response.status_code == 200
     served_model_preds = np.array(json.loads(scoring_response.content))
     np.testing.assert_array_equal(served_model_preds, model.predict(data, None))
+
+
+def test_mlflow_tracking_disabled_in_artifacts_only_mode():
+
+    port = get_safe_port()
+    cmd = ["mlflow", "server", "--port", str(port), "--artifacts-only"]
+    process = subprocess.Popen(cmd)
+    with pytest.raises(
+        HTTPError,
+        match="Endpoint disabled due to the mlflow server running in `--artifacts-only` mode.",
+    ):
+        _await_server_up_or_die(port, timeout=10)
+        resp = requests.get(f"http://localhost:{port}/api/2.0/mlflow/experiments/list")
+        augmented_raise_for_status(resp)
+    process.kill()
+
+
+def test_mlflow_artifact_list_in_artifacts_only_mode():
+
+    port = get_safe_port()
+    cmd = ["mlflow", "server", "--port", str(port), "--artifacts-only", "--serve-artifacts"]
+    process = subprocess.Popen(cmd)
+    try:
+        _await_server_up_or_die(port, timeout=10)
+        resp = requests.get(f"http://localhost:{port}/api/2.0/mlflow-artifacts/artifacts")
+        augmented_raise_for_status(resp)
+        assert resp.status_code == 200
+        assert resp.text == "{}"
+    finally:
+        process.kill()
+
+
+def test_mlflow_artifact_service_unavailable_without_config():
+
+    port = get_safe_port()
+    cmd = ["mlflow", "server", "--port", str(port)]
+    process = subprocess.Popen(cmd)
+    try:
+        _await_server_up_or_die(port, timeout=10)
+        resp = requests.get(f"http://localhost:{port}/api/2.0/mlflow-artifacts/artifacts")
+        assert resp.status_code == 503
```

Comment:
Can we check the error message as we do in `test_mlflow_tracking_disabled_in_artifacts_only_mode`?

Reply:
updated